DatadogCore/Tests/Datadog/TracerTests.swift

```diff
@@ -144,7 +144,7 @@ class TracerTests: XCTestCase {
         XCTAssertEqual(try spanMatcher.operationName(), "operation")
         XCTAssertEqual(try spanMatcher.startTime(), 1_576_404_000_000_000_000)
         XCTAssertEqual(try spanMatcher.duration(), 500_000_000)
-        XCTAssertEqual(try spanMatcher.dd.samplingRate(), 0.42)
+        XCTAssertEqual(try spanMatcher.dd.samplingRate(), 0.42, accuracy: 0.01)
```

Comment:
what is context behind accuracy here?

This should be well predictable value IMO.

Reply:
It is a fair call and may reveal an issue (not related to this PR though).

We changed from decoding `Float` to decoding `Double` in this PR:
```diff
-       func samplingRate() throws -> Float { try matcher.dd(forKeyPath: "_dd.agent_psr") }
+       func samplingRate() throws -> Double { try matcher.dd(forKeyPath: "_dd.agent_psr") }
```
The original value in this test is `42` (`Float`). It gets divided by `100` (`Float`) in implementation and serialized to JSON. I went deeper and I see it gets serialized as:
```
"_dd.agent_psr" = "0.4199999868869781";
```
hence by using `accuracy` we cover the actual problem. I'd go with `accuracy` to mitigate the problem, but the serialised precision is IMO worth opening a JIRA. WDYT @ganeshnj ?

Reply:
let's open a Jira ticket.

the original root cause is 🥁 🥁 🥁 

https://github.com/apple/swift-corelibs-foundation/issues/4255

Reply:
The issue in Swift stdlib is on "decoding", while we see it earlier in "encoding". I opened [RUM-3011](https://datadoghq.atlassian.net/browse/RUM-3011) with full context. Again, this is not a regression, so not blocking. Problem seems specific to iOS 16.2 rather than Xcode (Swift) version:
- 🐞 Xcode 15.0, iOS 16.2: "_dd.agent_psr":0.42500001192092896
- 🐞 Xcode 14.3.1, iOS 16.2: "_dd.agent_psr":0.42500001192092896
- ✅ Xcode 14.3.1, iOS 17.0: "_dd.agent_psr":0.425